querybook/server/lib/query_executor/executor_factory.py

```diff
@@ -33,16 +33,21 @@ def _get_executor_params_and_engine(query_execution_id, celery_task, session=Non
     if engine.deleted_at is not None:
         raise ArchivedQueryEngine("This query engine is disabled.")
 
+    proxy_user = user.username
+    executor_params = engine.get_engine_params()
+    if "proxy_user_id" in executor_params.keys():
+        try:
+            proxy_user = user.to_dict()[executor_params["proxy_user_id"]]
+        except KeyError:
```

Comment:
just throw on KeyError, otherwise this is going to be hard to debug

Reply:
Made changes to raise the same error when this occurs.